lib/source/tags.js

```diff
@@ -0,0 +1,101 @@
+/* global Log, Config */
+'use strict';
+
+const Crypto = require('crypto');
+const Aws = require('aws-sdk');
+
+const Source = require('./common');
+const Metadata = require('./metadata');
+const MetadataClient = require('../util/metadata-client');
+const Parser = require('./tags/parser');
+
+/**
+ * EC2 Tags Source
+ *
+ * Expose EC2 instance tags to the property-set
+ *
+ * @class Source.Tags
+ * @extends Source.Polling
+ *
+ * @param {Parser} parser
+ */
+class Tags extends Source.Polling(Parser) { // eslint-disable-line new-cap
+  /**
+   * Constructor
+   * @param {Object} opts
+   */
+  constructor(opts) {
+    // Inject defaults into options
+    const options = Object.assign({
+      timeout: Metadata.DEFAULT_TIMEOUT,
+      host: Metadata.DEFAULT_HOST
+    }, opts);
+
+    super('ec2-tags', options);
+
+    this._metadata = new MetadataClient(options);
+  }
+
+  /**
+   * Fetch implementation for EC2 tags API
+   * @param {Function} callback
+   * @private
+   */
+  _fetch(callback) {
+    const path = `/${Metadata.version}/dynamic/instance-identity/document`;
+
+    new Promise((resolve) => {
```

Comment:
Nope. The function that [calls _fetch()](https://github.com/rapid7/propsd/blob/master/lib/source/common.js#L333) doesn't do anything with a return value.